lib/puppet/rest/client.rb

```diff
@@ -31,43 +32,50 @@ def initialize(ssl_context:,
         @client.debug_dev = $stderr
       end
 
+      @ca_path = Puppet[:ssl_client_ca_auth] || Puppet[:localcacert]
+      @verifier = Puppet::SSL::Validator::DefaultValidator.new(@ca_path)
       configure_verify_mode(ssl_context)
 
       @dns_resolver = Puppet::Network::Resolver.new
     end
 
     # Make a GET request to the specified URL with the specified params.
-    # @param [String] url the full path to query
+    # @param [URI] url the full path to query
     # @param [Hash] query any URL params to add to send to the endpoint
     # @param [Hash] header any additional entries to add to the default header
     # @yields [String] chunks of the response body
     # @raise [Puppet::Rest::ResponseError] if the response status is not OK
     def get(url, query: nil, header: nil, &block)
       begin
-        @client.get_content(url, { query: query, header: header }) do |chunk|
+        @client.get_content(url.to_s, { query: query, header: header }) do |chunk|
           block.call(chunk)
         end
       rescue HTTPClient::BadResponseError => e
         raise Puppet::Rest::ResponseError.new(e.message, Puppet::Rest::Response.new(e.res))
+      rescue OpenSSL::OpenSSLError => e
+        Puppet::Util::SSL.handle_connection_error(e, @verifier, url.host)
       end
     end
 
     # Make a PUT request to the specified URL with the specified params.
-    # @param [String] url the full path to query
+    # @param [URI] url the full path to query
```

Comment:
I think this could be further specified to be `URI::HTTPS`; at least, that's the only thing we feed it from `Puppet::Rest::Route`.

Reply:
I updated this in my PR that I'm working on.